src/main/flight/imu.c

```diff
@@ -428,12 +428,17 @@ static void imuCalculateEstimatedAttitude(timeUs_t currentTimeUs)
     }
 #endif
 #if defined(USE_GPS)
-    else if (STATE(FIXED_WING) && sensors(SENSOR_GPS) && STATE(GPS_FIX) && gpsSol.numSat >= 5 && gpsSol.groundSpeed >= 300) {
+#if defined(USE_MAG)
+    else if
+#else
+    if
+#endif // USE_MAG
+            (STATE(FIXED_WING) && sensors(SENSOR_GPS) && STATE(GPS_FIX) && gpsSol.numSat >= 5 && gpsSol.groundSpeed >= 300) {
```

Comment:
Maybe add if (!useMag && ...
The useMag is only enabled when entering the previous if.

Reply:
That might be a better fix.  I was just trying to do a quick rearrange as I ran into this when I moved the `imuIsAccelerometerHealthy()` above that it was erroneously bein ghte `else if` of.  Let me look at it.

Reply:
Updated with simplified #define structure.